codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/protocols/RestJson1.kt

```diff
@@ -177,40 +188,47 @@ class RestJson1HttpSerializerGenerator(
                     val errorTrait = variantShape.expectTrait<ErrorTrait>()
                     val variantSymbol = symbolProvider.toSymbol(variantShape)
                     val data = safeName("var")
-                    val serializerSymbol =
-                        serde.member(symbolProvider.serializeFunctionName(variantShape))
-                    rustBlock("#TKind::${variantSymbol.name}($data) =>", errorSymbol) {
-                        rust(
-                            """
-                                #T(&mut object, &$data);
-                                object.key(${"code".dq()}).string(${httpBindingResolver.errorCode(variantShape).dq()});
-                            """.trimIndent(),
-                            serializerSymbol
-                        )
-                        if (variantShape.errorMessageMember() != null) {
+                    // TODO: the errorShapes members should probably be grabbed from the HTTP index as we do above in the responseSerializer.
+                    // I believe it will require a complete refactor of this method.
```

Comment:
maybe I'm missing something? But can't you just hoist the call here:
```
                        val bindings = httpBindingResolver.errorResponseBindings(it)
```
Up top? Then do something like 

`let members = bindings.filter { binding.location == HttpLocation.DOCUMENT }.map { it.member }`